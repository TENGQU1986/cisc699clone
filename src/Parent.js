import React, { useState, useEffect } from "react";
import Child from "./Child";
import ChildWithoutMemo from "./ChildWithoutMemo";
import "./App.css";
import Comments from "./Comments";
import LazyLoading from "./LazyLoading";

const Parent = ({ user }) => {
  const [counter, setCounter] = useState(0);
  const [input, setInput] = useState("");
  const [displayMemo, setDisplayMemo] = useState(false);
  const list = [1, 2, 3];

  useEffect(() => {
  }, []);

  const handleCounterClick = () => {
    setCounter(counter + 1);
  };

  const onInputChange = (e) => {
    const value = e.target.value;
    setInput(value);
  };

  const handleSwitch = () => {
    setDisplayMemo(!displayMemo);
  };
  return (
    <div className="App">
      <h1>Rendering Times Demo</h1>
      <span>hint: you can open developer tools to see the differentces</span>
      <div className="parent">
        <button onClick={handleSwitch} title="parent_button" className="button">
          {displayMemo ? "switch to unmemo" : "switch to memo"}
        </button>
        <h1>Hello from Parent: {counter}</h1>
        <div>
          <button onClick={handleCounterClick}>+1</button>
          <input
            type="text"
            onChange={onInputChange}
            title="parent_input"
            className="input"
          />
        </div>
      </div>
      {displayMemo ? (
        <Child counter={counter} list={list} />
      ) : (
        <ChildWithoutMemo counter={counter} list={list} />
      )}

    </div>
  );
};

export default Parent;
