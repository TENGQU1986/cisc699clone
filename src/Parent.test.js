import React from 'react';
import Parent from './Parent';
import { render, fireEvent, act } from "@testing-library/react";


it('should render Parent component', () => {
    const { container } = render(<Parent />);
    expect(container).toMatchSnapshot();
})

describe('clickButton', () => {
    it("on click", () => {
        const { queryByTitle } = render(<Parent />);
        const btn = queryByTitle("parent_button");
        expect(btn.innerHTML).toBe("switch to memo")
        fireEvent.click(btn);
        expect(btn.innerHTML).toBe("switch to unmemo")
        
    })

    it('on input change', async () => {
        const onInputChange = jest.fn();
        const { queryByTitle } = render(<Parent />);
        await act(async () => {
            fireEvent.input(await queryByTitle("parent_input"))
        })
        const input = await queryByTitle("parent_input");
        
        expect(Parent).toMatchSnapshot();
    })
})