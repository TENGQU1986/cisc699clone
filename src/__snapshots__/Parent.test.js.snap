// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`clickButton on input change 1`] = `[Function]`;

exports[`should render Parent component 1`] = `
<div>
  <div
    class="App"
  >
    <h1>
      Rendering Times Demo
    </h1>
    <span>
      hint: you can open developer tools to see the differentces
    </span>
    <div
      class="parent"
    >
      <button
        class="button"
        title="parent_button"
      >
        switch to memo
      </button>
      <h1>
        Hello from Parent: 
        0
      </h1>
      <div>
        <button>
          +1
        </button>
        <input
          class="input"
          title="parent_input"
          type="text"
        />
      </div>
    </div>
    <div
      class="Child"
    >
      <h1>
        Child without Memoization - 
        0
      </h1>
    </div>
  </div>
</div>
`;
